src/main/c/init.c

```diff
@@ -0,0 +1,29 @@
+#include "init.h"
+
+jclass peStatClass;
+jfieldID peStatClass_failedID;
+jfieldID peStatClass_lowID;
+jfieldID peStatClass_highID;
+jfieldID peStatClass_averageID;
+jfieldID peStatClass_stdID;
+
+
+#define FAIL_IF_NULL(x) if (!(x)) return JNI_ERR;
+
+jint JNI_OnLoad(JavaVM* vm, void* reserved) {
```

Comment:
I fear that this way of doing things will result in obscure errors that are impossible to trace.  This is because 1) you're doing it at library load time, and 2) you're swallowing the exceptions that would tell you what's going on.
You can reassure me that I'm wrong by performing the following test:  Change a field name or type in BwaMemPairEndStats, and then try to do some alignment.  If you get some obscure message about being unable to load the native library, then the test will have failed.
(In this case, you ought to do this initialization once, lazily, when you're first called to do an alignment so that you have a full environment, and can pass back specific and diagnostic exceptions.)

Reply:
:processResources
:classes
:compileTestJava
:processTestResources
:testClasses
:test

org.broadinstitute.hellbender.utils.bwa.BwaMemIndexTest.openIndex FAILED
    java.lang.NoSuchFieldError: failed_to_find_failed
        at java.lang.ClassLoader$NativeLibrary.load(Native Method)
        at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1941)
        at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1824)
        at java.lang.Runtime.load0(Runtime.java:809)
        at java.lang.System.load(System.java:1086)
        at org.broadinstitute.hellbender.utils.bwa.BwaMemIndex.loadNativeLibrary(BwaMemIndex.java:467)
        at org.broadinstitute.hellbender.utils.bwa.BwaMemIndex.createIndexImageFromIndexFiles(BwaMemIndex.java:121)
        at org.broadinstitute.hellbender.utils.bwa.BwaMemIndexTest.openIndex(BwaMemIndexTest.java:31)

Reply:
There is no need for lazy initialization. It works in OnLoad, I mean it does produce a easy to diagnose exception message.